de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/validation/WurstValidator.java

```diff
@@ -1960,11 +1979,33 @@ private void checkForDuplicatePackages(WurstModel model) {
     }
 
     private void checkBannedFunctions(ExprFunctionCall e) {
-        String[] banned = new String[]{
-                "TriggerRegisterVariableEvent" /* , "ExecuteFunc" */};
-        for (String name : banned) {
-            if (e.getFuncName().equals(name)) {
-                e.addError("The function " + name + " is not allowed in Wurst.");
+        if (e.getFuncName().equals("TriggerRegisterVariableEvent")) {
+            if (e.getArgs().size() > 1) {
+                if (e.getArgs().get(1) instanceof ExprStringVal) {
+                    ExprStringVal varName = (ExprStringVal) e.getArgs().get(1);
+                    TRVEHelper.protectedVariables.add(varName.getValS());
+                    WLogger.info("keep: " + varName.getValS());
+                    return;
+                } else if (e.getArgs().get(1) instanceof ExprVarAccess) {
+                    // Check if this is a two line hook... thanks Bribe
```

Comment:
Is this for a specific bj function? Maybe add a comment...

Reply:
This is for single nested calls iirc to detect them as statically resolvable. bribe used to use it in his damage engine.
```
function myfunc (with same params and return typ)
    TriggerRegisterVariableEvent(...)
```
A bit bad solution maybe :D